bindings/experimental/distrdf/python/DistRDF/Node.py

```diff
@@ -253,27 +253,43 @@ def npartitions(self, value):
     @property
     def nentries(self):
         """Retrieve the total number of entries from the dataset."""
-        return self._headnode.get_num_entries()
+        return self._headnode.nentries
 
     @property
     def treename(self):
         """Retrieve the name of the TTree."""
-        return self._headnode.get_treename()
+        try:
+            return self._headnode.treename
+        except AttributeError:
+            # Ugly, will be fixed when refactoring creation of Ranges in a separate commit
+            return None
 
     @property
     def tree(self):
         """Retrieve the TTree instance."""
-        return self._headnode.get_tree()
+        try:
+            return self._headnode.tree
+        except AttributeError:
+            # Ugly, will be fixed when refactoring creation of Ranges in a separate commit
+            return None
 
     @property
     def inputfiles(self):
         """Retrieve the input files of the dataset."""
-        return self._headnode.get_inputfiles()
+        try:
+            return self._headnode.inputfiles
+        except AttributeError:
+            # Ugly, will be fixed when refactoring creation of Ranges in a separate commit
+            return None
 
     @property
     def friendinfo(self):
         """Retrieve information about friend trees of the dataset."""
-        return self._headnode._get_friend_info()
+        try:
+            return self._headnode.friendinfo
+        except AttributeError:
+            # Ugly, will be fixed when refactoring creation of Ranges in a separate commit
+            return None
```

Comment:
as far as I can tell the only usage of the `RangesBuilder` class is at Node.py:632 where this is called:

```Python
RangesBuilder(self).build_ranges()
```

If this is the only way `RangesBuilder` is used, why does it need properties at all?

Reply:
Indeed, it doesn't. Since the RangesBuilder class will be removed in the next PR, I would leave it like this if it's ok with you. Otherwise for this PR I can put a check `if isinstance(self._headnode, TreeHeadNode)` in the RangesBuilder `__init__` function and set the right attributes accordingly

Reply:
we can leave it like this but please add a FIXME/TODO mentioning that `RangesBuilder` should probably just be a free function

Reply:
done!